SMSCounter.php

```diff
@@ -244,7 +259,23 @@ private function doCount($text, $supportShiftTables)
         }
 
         $messages = (int) ceil($length / $perMessage);
-        $remaining = ($perMessage * $messages) - $length;
+
+        if ($encoding === self::UTF16 && $length > $perMessage) {
+            $count = 0;
+            foreach ($unicodeArray as $char) {
+                if ($count === $perMessage) {
+                    $count = 0;
+                } elseif ($count > $perMessage) {
+                    $count = 2;
+                }
+
+                $count += $char >= 65536 ? 2 : 1;
+            }
+
+            $remaining = $perMessage - ($count > $perMessage ? 2 : $count);
```

Comment:
This will do return remaining chars in last part only of UTF16 encoded messages longer than 67